### ISSUE TITLE
xterm and display (imagemagick) in termux / Xserver

### ISSUE THREAD

**Issue by RalfWerner** [termux/x11-packages#49]
Hi Leonid

You wrote to me: "**Issues and bug reports must be posted on Github** in relevant repositories"
So I opened an issue here. You say twaik is responsible for xorg-server so here I limit my problems with the two standard applications to their representation with xsdl, mattkwan (android) and debian (linux). The mattkwan xserver app requires less than 1 MB, no setup (SDL/scale) in neccessary  and less than a second to start. Both servers **do not use** the Termux fonts.
![grafik](https://user-images.githubusercontent.com/45426380/50772452-2c55fe80-128e-11e9-82f4-8db44d9c8aca.png)

First: My goal is to use the functions of my apps in the PC environment with windows and mouse and on the phone with fingers and "shifts" (comparable to the Shell [n] in Termux). The client / server assignments should be exchangeable. On the phone then should be no Windows Manager necessary and the entire display be used. When connecting mouse and keyboard to the phone, they should work.

How should an Xserver work on the phone? Two examples that are "role models" for me (see videos in my dropbox):

1. TeamViewer (TV).
The TV remote control function allows editing a PC on a mobile phone. The zoom and the mobile phone rotation is implemented without reaction to the PC. This makes it easy to set the relevant work area. All mice / fingers (arrows) and keyboards of the participating computers (here three) are simultaneously displayed / synchronized and can be alternatively used - exampe video: https://www.dropbox.com/s/9sb1ft8ydw0u7mb/TVpad_phone.mp4?dl=0

2. Termux (with mc = midnight commander).
The zoom and rotation functions are well graded. Several windows (shells) can be edited in parallel. Most mc functions are to be executed by menus, function buttons and double (finger) click the keypad is rarely used. A distinction of mouse (finger) and cursor is not necessary / possible here, because there is no GUI. Zoom is also the scaling of the entire display with different fonts. About 20 of the 60 fonts also yield readable sizes.
Graphics / videos can be edited via their app's with double-click ("open"->am) in mc. In some (random) cases, the phone rotation is incorrectly implemented, which can usually be corrected by turning the keypad on/off. On "open" I get, as I wrote you an "open" error. The video: https://www.dropbox.com/s/lsyylstr9nsq3ml/Termux_mc.mp4?dl=0 was created with an older Termux installation.

I have installed the xterm and display packages on both systems and tested with some problems. I ignore the problems of missing window manegers, because typical android apps can do it without them. This results in the following 12 cases through client / server swap, two applications and 2 OS:

xterm linux->linux           xterm android->linux
xterm linux->mattkwan*) xterm android->mattkwan*)
xterm linux->xsdl           xterm android->xsdl

display linux->linux        display android->linux**)
display linux->mattkwan display android->mattkwan**)
display linux->xsdl         display android->xsdl**)

*) Error:
xterm terminates with mattkwan from linux with an abort and continued by android after an error. All others work. I have used the option -e mc, but on android _did not receive any quality comparable function to termux_mc_.

**) Error:
display on android shows the message despite the correct DISPLAY env:
**display: delegate library support not built-in '' (X11) @ error/display.c/DisplayImageCommand/1900.**
How can this error be corrected? **_What should be the option -display server?_**

Ps. When it's helpful and you tell my how to do, I could insert some sources/scripts, screenshots or videos in this issue. here an example
![grafik](https://user-images.githubusercontent.com/45426380/50772408-fa449c80-128d-11e9-800a-11d21c986e8c.png)

**Comment by xeffyr** [comment]
> How should an Xserver work on the phone?

> Two examples that are "role models"

You probably misunderstand what Xserver is. I suggest you to read https://en.wikipedia.org/wiki/X_Window_System.

The only role of Xserver is a *controlling of the video output*. Nothing else.
Generally, Xserver is not necessary in Termux as output can be done to the remote either via plain X11 proto or over X11 SSH forwarding.

> xterm terminates with mattkwan from linux with an abort and continued by android after an error.

First of all, **xterm** here is a shortcut for **aterm**. Xterm works really bad in Termux.
It it fails to launch, try to run it with graphical environment like fluxbox. **There no fix for that as some things like compositing available only with graphical environment.**

> How can this error be corrected? What should be the option -display server?

This option specifies a display socket. It can be, e.g. `-display :0` or `-display 127.0.0.1:0`. But usually setting the environment variable `DISPLAY` is sufficient.

**Comment by xeffyr** [comment]
Also, imagemagick is not compiled against x11 packages, so `display` command will not work.

**Comment by xeffyr** [comment]
In addition to everything I wrote above:

> On the phone then should be no Windows Manager necessary and the entire display be used.

1. Window manager already not necessary for running graphical programs.
2. Application can't use entire display when there no window manager. This behaviour cannot and won't be changed.

> When connecting mouse and keyboard to the phone, they should work.

They work. But they are handled by Android - another thing that cannot be changed. Termux is not operating system and can't manage hardware.

***

Since everything you requested going against __purpose__ and __implementation__ for Termux:X11 packages, I'm closing this issue as *wontfix*.

**Comment by xeffyr** [comment]
> Maybe we have different ideas about Xserver.

I'm talking about one from https://www.freedesktop.org/wiki/Software/xorg/. Nothing else as only this implementation will be used in x11-packages. Want something else - you are free to fork this repo.

> On Linux, the Xserver is typically "builtin".

This is the first time when I hear that on Linux Xserver is built-in.

However I know that is is **optional package** which is not required for running headless operating system.

> All three variants I have installed on my PC and was able to test the two examples described above and some of my own.

I don't know what you have installed and tested. But I prefer normal implementation that is widely used in Linux distributions.

> Did you compile imagemagick, how did you do that with display without X11 and did you insert the message */1900 (above)? Please tell me if you have a new imagemagick version!

The same way as [ALL](https://github.com/termux/x11-packages/tree/master/packages) of my packages are compiled. For example in Termux we have emacs without X support, but in this repository the package [emacs-x](https://github.com/termux/x11-packages/tree/master/packages/emacs-x) is available which provides emacs with X support.

> The management of the entire display is very simple: The first action of the Xserver is always XOpenDisplay. Thus, the mobile screen size is known to the client (runs eg. in a Termux shell) and can be used in XCreateWindow. 

I have no plans on modifying standard behaviour of the X server. All packages in this repository will stay close to original as much as possible.

Single-window output will break some programs that require multiple windows. So as I wrote in my previous message, **this is a wontfix**.

**Comment by xeffyr** [comment]
### Just for note:

This repository *just* provides utilities and libraries for creating graphical environment. Issues about *separate* and *third-party* applications used for graphical output are **not accepted here**.

If you want to ask for some feature in XSDL, mattkwan's Xserver or other application - ask in repository relevant to such applications and not here.
